tools/c7n_gcp/c7n_gcp/resources/mlengine.py

```diff
@@ -39,8 +43,100 @@ def get(client, event):
                 )})
 
 
+@MLModel.action_registry.register('set')
+class MLModelSet(MethodAction):
+    """The action is used for ML projects.models patch description field.
+
+    GCP action is https://cloud.google.com/ml-engine/reference/rest/v1/projects.models/patch
+
+    Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: ml-model-update-description
+            resource: gcp.ml-model
+            filters:
+              - type: value
+                key: name
+                value: projects/cloud-custodian/models/test
+            actions:
+              - type: set
+                description: Custom description
+                labels:
+                  - key: type
+                    value: new
+    """
+
+    schema = type_schema(
+        'set',
+        **{
+            'description': {
+                'type': 'string'
+            },
+            'labels': {
+                'type': 'array',
```

Comment:
a more natural form for adding labels is an object

ie: 

```
  labels:
     key: value
```

and then in jsonschema using patternProperties to ensure the value is a string.

Reply:
changed